test/db.tests.js

```diff
@@ -1228,6 +1228,37 @@ describe('LimitDBRedis', () => {
           });
         });
 
+        it('should return erl_quota_count >= 0 when ERL gets activated', (done) => {
+          params.erlQuota = { key: 'erlquotakey', per_cal_month: 10 };
+
+          // check erl not activated yet
+          redisExistsPromise(erlIsActiveKey)
+            .then((erlIsActiveExists) => assert.equal(erlIsActiveExists, 0))
+            // check erlQuotaKey does not exist
+            .then(() => redisExistsPromise(params.erlQuota.key))
+            .then((erlQuotaKeyExists) => assert.equal(erlQuotaKeyExists, 0))
+            // attempt to take elevated should work for first token
+            .then(() => takeElevatedPromise(params))
+            .then(() => redisExistsPromise(erlIsActiveKey))
+            .then((erlIsActiveKeyExists) => assert.equal(erlIsActiveKeyExists, 0))
+            // next takeElevated should activate ERL
+            .then(() => takeElevatedPromise(params))
+            .then((response) => assert.isTrue(response.erl_activated) && assert.isAtLeast(response.erl_quota_count, 0))
+            .then(() => done());
+        });
+
+        it('should return erl_quota_count = -1 when ERL had already been activated', (done) => {
```

Comment:
Actually, we would want the current remaining quota for reporting purposes, not the -1.

Reply:
When ERL gets activated it returns the quota. On the client side if erl_quota_count > 0 we know it's been activated in this call and the value of the quota.

On subsequent calls, if ERL is already activated (within the erl_activation_period), we will receive as response `erl_activated=true` and `erl_quota_count=-1`. 

Otherwise we would need a third parameter indicating if ERL flipped its value or not.

Reply:
For instance:
- Redis status: `erlActivationKey = 0`
  - `erl_activated=false`
  - `erl_quota_count=-1`
- Redis status: `erlActivationKey = 1` => this call activated ERL
  - `erl_activated=true`
  - `erl_quota_count=9`
- Redis status: `erlActivationKey = 1` => this call encountered ERL already activated
  - `erl_activated=true`
  - `erl_quota_count=-1`

Reply:
Ohh because consequent calls while erl is already active doesn't use from the quota. Gotcha.